docs/source/design/dataset_argument.rst

```diff
@@ -0,0 +1,62 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_dataset_argument:
+
+********************
+``dataset`` argument
+********************
+
+.. topic:: Specification scope and status
+
+   This specification describes the current implementation.
+
+All commands which operate on datasets have a ``dataset`` argument (``-d`` or
+``--dataset`` for the command line API) to identify a single dataset as the
+context of an operation.
+
+
+Impact on relative path resolution
+==================================
+
+With one exception, the nature of a provided ``dataset`` argument does **not**
+impact the interpretation of relative paths. Relative paths are always considered
+to be relative to the process working directory.
+
+The one exception to this rule is passing a ``Dataset`` object instance as
+``dataset`` argument value in the Python API. In this, and only this, case, a
+relative path is interpreted as relative to the root of the respective dataset.
+
+
+Special values
+==============
+
+There are some pre-defined "shortcut" values for dataset arguments:
+
+``^``
+   Represents to the topmost superdataset that contains the dataset the current
+   directory is part of.
+``^.``
+   Represents the root directory of the dataset the current directory is part of.
+``///``
+   Represents the "default" dataset located under `$HOME/datalad/`.
+
+
+Use cases
+=========
+
+Save modification in superdataset hierarchy
+-------------------------------------------
+
+Sometimes it is convenient to work only in the context of a subdataset.
+Executing a ``datalad save <subdataset content>`` will record changes to the
+subdataset, but will leave existing superdatasets dirty, as the subdataset
+state change will not be saved there. Using the ``dataset`` argument it is
```

Comment:
echoing resolved discussion with @bpoldrack - I think the "context of operation" is still to vague, and there it seems to suggest something like 

```rst
Some (or all?) operations can "transfer" the context of operation to the subdatasets provided in the path, if no `--dataset` argument is provided.
```
which would possibly provide explanation on how come `save <subdataset content>` does not operate in the context of current dataset (which it might if one of the paths is within this dataset) but "in the context of a subdataset".

Actual reality is actually even a bit more tricky, since 

<details>
<summary>we do also operate in the context of CWD dataset - since we do `save` the change saved within subdataset </summary> 

```shell
(git-annex)lena:/tmp/testds[master]git
$> datalad status
untracked: datasets.datalad.org (directory)
 modified: hcp-100206-git-annex2 (dataset)
(dev3) 1 26057.....................................:Thu 05 Aug 2021 10:43:32 AM EDT:.
(git-annex)lena:/tmp/testds[master]git
$> touch sub1/newfile newfilehere1 newfilehere2
(dev3) 1 26058.....................................:Thu 05 Aug 2021 10:43:53 AM EDT:.
(git-annex)lena:/tmp/testds[master]git
$> datalad status
untracked: datasets.datalad.org (directory)
untracked: newfilehere1 (file)
untracked: newfilehere2 (file)
 modified: hcp-100206-git-annex2 (dataset)
 modified: sub1 (dataset)
(dev3) 1 26059.....................................:Thu 05 Aug 2021 10:43:59 AM EDT:.
(git-annex)lena:/tmp/testds[master]git
$> datalad save sub1/newfile newfilehere1
add(ok): newfile (file)                                                                                                                                                                                              
save(ok): sub1 (dataset)                                                                                                                                                                                             
add(ok): sub1 (file)                                                                                                                                                                                                 
add(ok): newfilehere1 (file)                                                                                                                                                                                         
save(ok): . (dataset)                                                                                                                                                                                                
action summary:                                                                                                                                                                                                      
  add (ok: 3)
  save (ok: 2)
(dev3) 1 26060.....................................:Thu 05 Aug 2021 10:44:11 AM EDT:.
(git-annex)lena:/tmp/testds[master]git
$> datalad status
untracked: datasets.datalad.org (directory)
untracked: newfilehere2 (file)
 modified: hcp-100206-git-annex2 (dataset)

(git-annex)lena:/tmp/testds[master]git
$> touch sub1/newfile2                          
(dev3) 1 26062.....................................:Thu 05 Aug 2021 10:45:06 AM EDT:.
(git-annex)lena:/tmp/testds[master]git
$> datalad save sub1/newfile2
add(ok): newfile2 (file)                                                                                                                                                                                             
save(ok): sub1 (dataset)                                                                                                                                                                                             
action summary:                                                                                                                                                                                                      
  add (ok: 1)
  save (ok: 1)

$> datalad --version
datalad 0.14.7+1.g721315d5d.dirty

```
</details>

so may be this statement is not correct altogether?

Reply:
I think I just got mislead by `save(ok): sub1 (dataset)` assuming that it was to inform that we saved updated state of `sub1` within current dataset

<details>
<summary>but it is not the case</summary> 

```shell
(git-annex)lena:/tmp/testds[master]git-annex
$> git status       
On branch master
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   sub1 (new commits)

no changes added to commit (use "git add" and/or "git commit -a")
(dev3) 1 26407 [1].....................................:Fri 06 Aug 2021 09:25:21 AM EDT:.
(git-annex)lena:/tmp/testds[master]git-annex
$> datalad save sub1/
(dev3) 1 26408 [1].....................................:Fri 06 Aug 2021 09:25:42 AM EDT:.
(git-annex)lena:/tmp/testds[master]git-annex
$> touch sub1/newfile3
(dev3) 1 26409 [1].....................................:Fri 06 Aug 2021 09:25:48 AM EDT:.
(git-annex)lena:/tmp/testds[master]git-annex
$> datalad save sub1  
add(ok): newfile3 (file)                                                                                  
save(ok): sub1 (dataset)                                                                                  
action summary:                                                                                           
  add (ok: 1)
  save (ok: 1)
(dev3) 1 26410 [1].....................................:Fri 06 Aug 2021 09:25:55 AM EDT:.
(git-annex)lena:/tmp/testds[master]git-annex
$> git status         
On branch master
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   sub1 (new commits)

no changes added to commit (use "git add" and/or "git commit -a")
(dev3) 1 26411 [1].....................................:Fri 06 Aug 2021 09:26:21 AM EDT:.
(git-annex)lena:/tmp/testds[master]git-annex
$> git show | head
commit d7e5959365f6bc575a89e1a49d8dc1d504680c6f
Author: Yaroslav Halchenko <debian@onerussian.com>
Date:   Thu Aug 5 10:44:11 2021 -0400

    [DATALAD] Recorded changes

diff --git a/newfilehere1 b/newfilehere1
new file mode 120000
index 0000000..ce38032
--- /dev/null
(dev3) 1 26412 [1].....................................:Fri 06 Aug 2021 09:26:41 AM EDT:.
(git-annex)lena:/tmp/testds[master]git-annex
$> touch sub1/newfile4                                             
(dev3) 1 26413 [1].....................................:Fri 06 Aug 2021 09:28:09 AM EDT:.
(git-annex)lena:/tmp/testds[master]git-annex
$> datalad save sub1/      
add(ok): newfile4 (file)                                                                                  
save(ok): sub1 (dataset)                                                                                  
action summary:                                                                                           
  add (ok: 1)
  save (ok: 1)
(dev3) 1 26414 [1].....................................:Fri 06 Aug 2021 09:28:28 AM EDT:.
(git-annex)lena:/tmp/testds[master]git-annex
$> touch sub1/newfile5
(dev3) 1 26415 [1].....................................:Fri 06 Aug 2021 09:28:31 AM EDT:.
(git-annex)lena:/tmp/testds[master]git-annex
$> datalad save sub1/newfile5
add(ok): newfile5 (file)                                                                                  
save(ok): sub1 (dataset)                                                                                  
action summary:                                                                                           
  add (ok: 1)
  save (ok: 1)
(dev3) 1 26416 [1].....................................:Fri 06 Aug 2021 09:28:38 AM EDT:.
(git-annex)lena:/tmp/testds[master]git-annex
$> git status
On branch master
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   sub1 (new commits)

no changes added to commit (use "git add" and/or "git commit -a")

```
</details>

so the statement was correct.

Reply:
I will mark it as resolved and will not bother about clarifying "context" here